src/AreaChart.js

```diff
@@ -0,0 +1,60 @@
+import React from 'react';
+const {PropTypes} = React;
+import _ from 'lodash';
+import d3 from 'd3';
+
+import {makeAccessor, domainFromData, combineDomains} from './utils/Data';
+import * as CustomPropTypes from './utils/CustomPropTypes';
+
+// AreaChart represents a simple bivariate area chart,
+// a filled path drawn between two lines (datasets).
+
+// todo horizontal prop, for filling area horizontally?
+// todo support categorical data?
+// todo support passing 2 data arrays and generating area between them? d3 doesn't seem to support this
+// todo build StackedAreaChart that composes multiple AreaCharts
+
+export default class AreaChart extends React.Component {
+  static propTypes = {
+    // the array of data objects
+    data: PropTypes.array.isRequired,
+    // accessors for X & Y coordinates
+    getX: CustomPropTypes.getter,
+    getY: CustomPropTypes.getter,
+    getYEnd: CustomPropTypes.getter,
+    // style applied to path element
+    pathStyle: PropTypes.object,
+
+    scaleType: PropTypes.object,
+    scale: PropTypes.object,
+  };
+
+  static getDomain(props) {
```

Comment:
Yep exactly @thehindenburg - the parent (XYPlot) calls this *before* rendering itself, because it needs to know the domains of all of its children before render, which is why it's a static. Similar patterns exist for margin and scaleType. We should discuss this in depth sometime soon and see if there are any good alternatives... But I haven't found them :)